gnovm/pkg/gnolang/stacktrace.go

```diff
@@ -0,0 +1,81 @@
+package gnolang
+
+import (
+	"fmt"
+	"strings"
+)
+
+const maxStacktraceSize = 128
+
+type Execution struct {
+	Stmt  Stmt
+	Frame *Frame
+}
+type Stacktrace struct {
+	Executions      []Execution
+	NumFramesElided int
+}
+
+func (s Stacktrace) String() string {
+	var builder strings.Builder
+
+	for i, e := range s.Executions {
+		if s.NumFramesElided > 0 && i == maxStacktraceSize/2 {
+			builder.WriteString(fmt.Sprintf("...%d frame(s) elided...\n", s.NumFramesElided))
+		}
+
+		switch {
+		case e.Stmt == nil:
+			builder.WriteString(fmt.Sprintf("%s()\n", e.Frame.Func))
```

Comment:
Maybe use `fmt.Fprintf(builder, ...)`

Reply:
Yes right, it is better.

Done here: https://github.com/gnolang/gno/pull/2145/commits/76faf71ac6c1232b8b59016c23db918810486f83